tests/acceptance/features/bootstrap/TusContext.php

```diff
@@ -74,59 +74,63 @@ public function before(BeforeScenarioScope $scope): void {
 	 * @throws GuzzleException
 	 */
 	public function uploadFileViaTus(string $user, string $resource, string $spaceName): void {
-		$resourceLocation = $this->getResourceLocation($user, $resource, $spaceName);
-		$file = \fopen($this->acceptanceTestsDirLocation() . $resource, 'r');
-
-		$this->featureContext->setResponse(
-			HttpRequestHelper::sendRequest(
-				$resourceLocation,
-				"",
-				'HEAD',
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				[],
-				""
-			)
-		);
-		$this->featureContext->theHTTPStatusCodeShouldBe(200, "Expected response status code should be 200");
-
-		$this->featureContext->setResponse(
-			HttpRequestHelper::sendRequest(
-				$resourceLocation,
-				"",
-				'PATCH',
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				["Tus-Resumable" => "1.0.0", "Upload-Offset" => 0, 'Content-Type' => 'application/offset+octet-stream'],
-				$file
-			)
-		);
+		$this->userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi($user, $resource, $spaceName);
 		$this->featureContext->theHTTPStatusCodeShouldBe(204, "Expected response status code should be 204");
 	}
+	
+	/**
+	 * @When /^user "([^"]*)" uploads a file "([^"]*)" via TUS inside of the space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi(string $user, string $resource, string $spaceName) {
+		$this->createTusResource($user, $resource, $spaceName);
+		$this->featureContext->theHTTPStatusCodeShouldBe(201, "Expected response status code should be 201");
```

Comment:
These two steps can be separately used as a single given step:
```
Given user "" has created a TUS resource for "" inside space ""
```
a upload when step only doing the upload would be good.